### PR TITLE
feat(calcs): for labelmap statistics

### PR DESCRIPTION
**Pull request by wayfarer3130**
Fixes https://github.com/cornerstonejs/cornerstone3D/issues/1628

This pull request includes several changes to the `common/reviews/api/tools.api.md` and `packages/tools` files, focusing on enhancing the functionality of tools and segmentation strategies. The most important changes include adding new methods and parameters, updating configurations, and improving segmentation event handling.

Enhancements to tools and segmentation strategies:

* [`common/reviews/api/tools.api.md`](diffhunk://#diff-2403182f0aa88e43f1f8fcd59594bbd3a62cac992598d1ab80e4e0f1bdd0628eR909-R910): Added `getStatistics` method to `BrushTool`, introduced `VolumetricCalculator` class, and modified `applyActiveStrategyCallback` to accept extra arguments. [[1]](diffhunk://#diff-2403182f0aa88e43f1f8fcd59594bbd3a62cac992598d1ab80e4e0f1bdd0628eR909-R910) [[2]](diffhunk://#diff-2403182f0aa88e43f1f8fcd59594bbd3a62cac992598d1ab80e4e0f1bdd0628eR4865) [[3]](diffhunk://#diff-2403182f0aa88e43f1f8fcd59594bbd3a62cac992598d1ab80e4e0f1bdd0628eL648-R648)
* [`packages/tools/src/tools/segmentation/BrushTool.ts`](diffhunk://#diff-b7dd1f228b12fbe8dbadc84bc006af903cdd0425e7e0262ded0b7ca624bb0505R115-R143): Added `getStatistics` method and new strategies for thresholding inside spheres with island removal. [[1]](diffhunk://#diff-b7dd1f228b12fbe8dbadc84bc006af903cdd0425e7e0262ded0b7ca624bb0505R115-R143) [[2]](diffhunk://#diff-b7dd1f228b12fbe8dbadc84bc006af903cdd0425e7e0262ded0b7ca624bb0505R697-R711)
* [`packages/tools/src/stateManagement/segmentation/events/triggerSegmentationDataModified.ts`](diffhunk://#diff-e069326172939ab34772a5a526be02d59ec46a8de1c76ea6b8124889d3902167L13-R19): Updated `triggerSegmentationDataModified` to include `segmentIndex` parameter.

Configuration and event handling improvements:

* [`packages/tools/src/enums/StrategyCallbacks.ts`](diffhunk://#diff-050c272cceb49b679f75fa75837eab77ee9d08dd445ee41ba95af81e88059150R53-R55): Added `GetStatistics` callback to `StrategyCallbacks` enum.
* [`packages/tools/src/tools/segmentation/strategies/BrushStrategy.ts`](diffhunk://#diff-31ecdd87eeedef42064215477d1f402c69236c8e203f4674335aab5da81da9afR116-R118): Enhanced `BrushStrategy` to handle `GetStatistics` callback and pass additional arguments. [[1]](diffhunk://#diff-31ecdd87eeedef42064215477d1f402c69236c8e203f4674335aab5da81da9afR116-R118) [[2]](diffhunk://#diff-31ecdd87eeedef42064215477d1f402c69236c8e203f4674335aab5da81da9afL380-R392)

Example updates:

* [`packages/tools/examples/labelmapSegmentationDynamicThreshold/index.ts`](diffhunk://#diff-ee3119b5b6d5d1310e7258ca5e5dd494f634f31883b715dd54b026f94ec6df9bR93-R154): Updated example to include new tools and configurations, and improved the segmentation representation handling. [[1]](diffhunk://#diff-ee3119b5b6d5d1310e7258ca5e5dd494f634f31883b715dd54b026f94ec6df9bR93-R154) [[2]](diffhunk://#diff-ee3119b5b6d5d1310e7258ca5e5dd494f634f31883b715dd54b026f94ec6df9bL191-R279)

### CODE REVIEW

File: packages/tools/src/utilities/math/basic/BasicStatsCalculator.ts [hunk 1]
```diff
@@ -14,6 +14,8 @@ export default class BasicStatsCalculator extends Calculator {
   // Values for Welford's algorithm
   private static runMean = [0];
   private static m2 = [0];
+  private static squaredDiffSum = [0];
```
Comment (sedghi) [inline packages/tools/src/utilities/math/basic/BasicStatsCalculator.ts:17]: @wayfarer3130 can you double check this file only? i think the rest is good to go

**Review by sedghi — APPROVED** [review]
Looks great

File: packages/tools/src/utilities/math/basic/BasicStatsCalculator.ts [hunk 2]
```diff
@@ -36,13 +38,16 @@ export default class BasicStatsCalculator extends Calculator {
     ) {
       this.max.push(this.max[0], this.max[0]);
       this.min.push(this.min[0], this.min[0]);
+      this.min.push(this.min[0], this.min[0]);
```
Comment (wayfarer3130) [inline packages/tools/src/utilities/math/basic/BasicStatsCalculator.ts]: Is there a reason to duplicate the min value?

File: packages/tools/src/utilities/segmentation/VolumetricCalculator.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  public static getStatistics(options: {
+    spacing?: number;
+    unit?: string;
+  }): NamedStatistics {
+    const { spacing } = options;
+    // Get the basic units
+    const stats = BasicStatsCalculator.getStatistics();
+
+    // Add the volumetric units
+    const volumeUnit = spacing ? 'mm\xb3' : 'voxels\xb3';
```
Comment (wayfarer3130) [inline packages/tools/src/utilities/segmentation/VolumetricCalculator.ts:18]: This should be ml, so off by a factor of 1000

**Pull request merged**